src/dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -395,15 +396,32 @@ Command StdIO()
                         kernel.UseQuitCommand();
                         
                         var kernelServer = kernel.CreateKernelServer(startupOptions.WorkingDir);
+                        var frontEndKernel = kernelServer.GetFrontEndKernel("vscode");
+                        kernel.Add(frontEndKernel);
 
                         if (startupOptions.EnableHttpApi)
                         {
                             var clientSideKernelClient =new SignalRBackchannelKernelClient();
 
                             if (isVsCode)
                             {
-                                services.AddSingleton(clientSideKernelClient);
+                                kernel.VisitSubkernels(k =>
+                                {
+                                    switch (k)
+                                    {
+                                        case CSharpKernel ck:
+                                            ck.UseVSCodeHelpers();
+                                            break;
+                                        case FSharpKernel fk:
+                                            fk.UseVSCodeHelpers();
+                                            break;
+                                    }
+                                });
+                                // TODO: move this
+                                kernel.RegisterCommandType<GetInput>();
```

Comment:
The fact that we're doing this here means that for embedded kernels, these gestures won't work.

I wonder if a more declarative approach to command and event serialization registration would be easier at this point.

Reply:
Moved to a more appropriate location.